src/basic/os-util.c

```diff
@@ -58,6 +58,28 @@ int path_is_extension_tree(const char *path, const char *extension, bool relax_e
         return 1;
 }
 
+int path_is_syscfg_tree(const char *path, const char *syscfg, bool relax_syscfg_release_check) {
+        int r;
+
+        assert(path);
+
+        /* Does the path exist at all? If not, generate an error immediately. This is useful so that a missing root dir
+         * always results in -ENOENT, and we can properly distinguish the case where the whole root doesn't exist from
+         * the case where just the os-release file is missing. */
+        if (laccess(path, F_OK) < 0)
+                return -errno;
+
+        /* We use /etc/syscfg-release.d/syscfg-release[.NAME] as flag for something being a system configuration,
+         * and {/etc|/usr/lib}/os-release as a flag for something being an OS (when not an extension). */
+        r = open_syscfg_release(path, syscfg, relax_syscfg_release_check, NULL, NULL);
+        if (r == -ENOENT) /* We got nothing */
+                return 0;
+        if (r < 0)
+                return r;
+
+        return 1;
+}
```

Comment:
please don't copy/paste all this code. i.e. map these two functions onto the same internal function that is just parameterized slightly differently, i.e. the file basename to use in the paths as string (i.e. "sysext-release" or "extension-release"), and the xattr name to check for, and similar.